src/client/dfuse/il/int_posix.c

```diff
@@ -364,39 +364,7 @@ ioil_fini(void)
 	struct ioil_pool *pool, *pnext;
 	struct ioil_cont *cont, *cnext;
 	int               rc;
-	pid_t             tid = syscall(SYS_gettid);
-
-	if (tid != ioil_iog.iog_init_tid) {
-		DFUSE_TRA_INFO(&ioil_iog, "Ignoring destructor from alternate thread");
-		return;
-	}
-
-	ioil_iog.iog_initialized = false;
-
-	DFUSE_TRA_DOWN(&ioil_iog);
-	vector_destroy(&fd_table);
-
-	ioil_show_summary();
-
-	/* Tidy up any open connections */
-	d_list_for_each_entry_safe(pool, pnext,
-				   &ioil_iog.iog_pools_head, iop_pools) {
-		d_list_for_each_entry_safe(cont, cnext,
-					   &pool->iop_container_head,
-					   ioc_containers) {
-			/* Retry disconnect on out of memory errors, this is mainly for fault
-			 * injection testing.  Do not attempt to shrink the pool here as that
-			 * is tried later, and if the container close succeeds but pool close
-			 * fails the cont may not be valid afterwards.
-			 */
-			rc = ioil_shrink_cont(cont, false, true);
-			if (rc == -DER_NOMEM)
-				ioil_shrink_cont(cont, false, true);
-		}
-		rc = ioil_shrink_pool(pool);
-		if (rc == -DER_NOMEM)
-			ioil_shrink_pool(pool);
-	}
+	pid_t             pid;
```

Comment:
unfortunately this is still wrong. you are destroying all EQs for threads that are now in a shared pool and can be in-use by other threads!
Im talking about the no-fork case and just using threads.

Reply:
Thank you! You are right. It would need more work.